src/Autoloader.php

```diff
@@ -54,25 +59,39 @@ public static function load()
         ));
     }
 
-    /**
-     * @return ClassLoader
-     * @throws RuntimeException
-     */
-    private static function getClassLoader()
+    private static function getClassLoader(): ?ClassLoader
     {
         if (getenv('COMPOSER_VENDOR_DIR') && file_exists(getenv('COMPOSER_VENDOR_DIR') . '/autoload.php')) {
-            return include getenv('COMPOSER_VENDOR_DIR') . '/autoload.php';
+            /** @psalm-suppress MixedAssignment */
+            $loader = include getenv('COMPOSER_VENDOR_DIR') . '/autoload.php';
+            if (!$loader instanceof ClassLoader) {
+                return null;
+            }
+
+            return $loader;
         }
 
         if (file_exists(__DIR__ . '/../../../autoload.php')) {
-            return include __DIR__ . '/../../../autoload.php';
+            /** @psalm-suppress MixedAssignment */
+            $loader = include __DIR__ . '/../../../autoload.php';
+            if (!$loader instanceof ClassLoader) {
+                return null;
+            }
+
+            return $loader;
         }
 
         if (file_exists(__DIR__ . '/../vendor/autoload.php')) {
-            return include __DIR__ . '/../vendor/autoload.php';
+            /** @psalm-suppress MixedAssignment */
+            $loader = include __DIR__ . '/../vendor/autoload.php';
+            if (!$loader instanceof ClassLoader) {
```

Comment:
If we invert this conditional, we can get rid of one return statement too

Reply:
We don't have a test scenario for when this returns something different than the class loader from composer 🤔

Reply:
Psalm is not able to infer the type and thus I prefer doing explicit checks here.

Reply:
I could do an assertion instead if thats cool?

Reply:
I actually prefer `instsanceOf` over `assert` due to the fact that the file might be (whyever) removed between `file_exists` and `include` and thus it could return `false`.

Switching to `require` would be more immersive and thus to be more failproof, I'll keep it as is.

Reply:
> If we invert this conditional, we can get rid of one return statement too

I meant something like this: 

```php
    if ($loader instanceof ClassLoader) {
        return $loader;
    }
    
    return null;
}
```

 > We don't have a test scenario for when this returns something different than the class loader from composer thinking

As in unit/integration tests covering this path

Reply:
Ah, I see that the code has been simplified: all good :D